src/main/java/crystal/Parser.java

```diff
@@ -0,0 +1,74 @@
+package crystal;
+
+import crystal.command.Command;
+import crystal.command.ListCommand;
+import crystal.command.UnmarkCommand;
+import crystal.command.MarkCommand;
+import crystal.command.TodoCommand;
+import crystal.command.DeadlineCommand;
+import crystal.command.EventCommand;
+import crystal.command.ByeCommand;
+import crystal.command.DeleteCommand;
+import crystal.command.FindCommand;
+
+/**
+ * Represents the Parser task.
+ *
+ */
+
+public class Parser {
+
+    /**
+     * Returns the respective command object based on the command
+     * the user enters.
+     *
+     * @param command User command
+     * @throws CrystalException When user command is not recognised
+     */
+    public static Command parse(String command) throws CrystalException {
```

Comment:
Your `parse` method seems too long to me... Maybe you would like to try splitting them out w.r.t. the `command`? As suggestion, maybe a `handleCommand` or `parseCommand` private method can be introduced to increase the readability of this method?

Reply:
Ok, i will try it out!